### PR TITLE
JS Datadog logger

### PR DESCRIPTION
**Pull request by benrudolph**
@dimagi/js-team  this should get our feet wet in terms of JS error tracking. curious to see what kind of volume we get on this. and this won't blow up our emails like last time @czue :smile: this'll output this log format when your logging config is setup correctly (requires an ansible bit):

```
jserror.count 1450303032.07 1 metric_type=counter url=http://localhost:8000/a/*/dashboard/project/ bot=False browser_name=Chrome browser_version=46.0.2490.86 file=None os=Macintosh
```

which datadog should be able to parse correctly

### CODE REVIEW

**Comment by snopoke** [comment]
nice @benrudolph. not mergable

**Comment by esoergel** [comment]
Looks great!  I'd love to see a demo of this once it's live and we have some error data.

**Comment by benrudolph** [comment]
cool, will update. @snopoke when i was doing this i found that there was already a logger called `datadog` but i couldn't find it in our localsettings on ansible. just thought i'd give you a heads up in case you wanted that data: https://github.com/dimagi/commcare-hq/blob/ddog-logger/corehq/util/datadog/__init__.py#L5

**Comment by snopoke** [comment]
@benrudolph the datadog logger I was using was just for generic datadog related logging. It doesn't need any special logging config since its either for local usage or for logging exceptions / errors

**Comment by snopoke** [comment]
:boot: :construction:

**Comment by benrudolph** [comment]
:white_check_mark:

File: corehq/apps/hqwebapp/views.py [hunk 1]
```diff
@@ -481,27 +484,26 @@ def debug_notify(request):
...
+    if agent:
+        parsed_agent = httpagentparser.detect(agent)
+        os = browser_name = browser_version = DATADOG_UNKNOWN
+        bot = parsed_agent.get('bot', False)
+        if 'os' in parsed_agent:
+            os = parsed_agent['os'].get('name', DATADOG_UNKNOWN)
+
+        if 'browser' in parsed_agent:
+            browser_version = parsed_agent['browser'].get('version', DATADOG_UNKNOWN)
+            browser_name = parsed_agent['browser'].get('name', DATADOG_UNKNOWN)
```
Comment (dannyroberts) [inline corehq/apps/hqwebapp/views.py]: Is the nesting wrong maybe? Looks like if agent is falsy, you'll get down to the `log_counter` and `os`, `browser_version`, `browser_name`, and `bot` won't be defined.

Comment (benrudolph) [inline corehq/apps/hqwebapp/views.py]: :eyes: you have a point

Comment (benrudolph) [inline corehq/apps/hqwebapp/views.py]: https://github.com/dimagi/commcare-hq/pull/9667

**Comment by benrudolph** [comment]
@esoergel this is up and running and has some data:

https://app.datadoghq.com/dash/90368/javascript-error-monitoring?live=true&page=0&is_auto=false&from_ts=1452097379583&to_ts=1452183779583&tile_size=m

**Comment by czue** [comment]
slick! and not as many as I expected!

**Pull request merged**